src/chainrule.jl

```diff
@@ -0,0 +1,23 @@
+### Projecting a tuple to SMatrix leads to ChainRulesCore._projection_mismatch by default, so overloaded here
+function (project::ChainRulesCore.ProjectTo{<:Tangent{<:Tuple}})(dx::SArray)
+    dy = reshape(dx, axes(project.elements))  # allows for dx::OffsetArray
+    dz = ntuple(i -> project.elements[i](dy[i]), length(project.elements))
+    return ChainRulesCore.project_type(project)(dz...)
+end
+
+### Project SArray to SArray
+function ChainRulesCore.ProjectTo(x::SArray{S,T}) where {S, T}
+    return ChainRulesCore.ProjectTo{SArray}(; element=ChainRulesCore._eltype_projectto(T), axes=S)
```

Comment:
If anyone is relying on `ProjectTo(x).axes` actually containing `axes(x)`, then this will be very surprising. Maybe this `Tuple{size...}` thing ought to have a different name:

```suggestion
    return ChainRulesCore.ProjectTo{SArray}(; element=ChainRulesCore._eltype_projectto(T), axes=axes(x), static_size=S)
```